node/src/cli/cli_parachain.rs

```diff
@@ -224,7 +224,7 @@ impl sc_cli::SubstrateCli for RelayChainCli {
 #[derive(Debug, Parser)]
 pub struct ExportGenesisStateCommand {
```

Comment:
We should be able to remove custom `ExportGenesisStateCommand` subcommand struct, because it is included in cumulus now:
https://github.com/substrate-developer-hub/substrate-parachain-template/compare/polkadot-v0.9.23...polkadot-v0.9.26#diff-76397c90c6ee2b6b98efd989d3ae3c11e0bde3db30b5ef46f2762180f56041a8

In *node/cli.rs*, we can then replace:
```rust
    /// Export the genesis state of the parachain.
    #[cfg(feature = "parachain")]
    #[clap(name = "export-genesis-state")]
    ExportGenesisState(cli_parachain::ExportGenesisStateCommand),

    /// Export the genesis wasm of the parachain.
    #[cfg(feature = "parachain")]
    #[clap(name = "export-genesis-wasm")]
    ExportGenesisWasm(cli_parachain::ExportGenesisWasmCommand),
```

by
```rust
    /// Export the genesis state of the parachain.
    #[cfg(feature = "parachain")]
    ExportGenesisState(cumulus_client_cli::ExportGenesisStateCommand),

    /// Export the genesis wasm of the parachain.
    #[cfg(feature = "parachain")]
    ExportGenesisWasm(cumulus_client_cli::ExportGenesisWasmCommand),
```

See 
https://github.com/substrate-developer-hub/substrate-parachain-template/compare/polkadot-v0.9.23...polkadot-v0.9.26#diff-76397c90c6ee2b6b98efd989d3ae3c11e0bde3db30b5ef46f2762180f56041a8L46-L78 
and
https://github.com/substrate-developer-hub/substrate-parachain-template/compare/polkadot-v0.9.23...polkadot-v0.9.26#diff-76397c90c6ee2b6b98efd989d3ae3c11e0bde3db30b5ef46f2762180f56041a8R27-R31

Reply:
Also in line 53 we can parse the command directly using `clap` instead of parsing functions exposed by polkadot client `RunCmd`: https://github.com/substrate-developer-hub/substrate-parachain-template/compare/polkadot-v0.9.23...polkadot-v0.9.26#diff-76397c90c6ee2b6b98efd989d3ae3c11e0bde3db30b5ef46f2762180f56041a8R91

Reply:
I don't think we can remove completely as we are taking parachain_id as parameter and that goes to following code to fill details in `Extensions` struct.
```rust
32 pub fn battery_station_staging_config(                                                                                                                                                                                                                                                                                   
 31     #[cfg(feature = "parachain")] parachain_id: cumulus_primitives_core::ParaId,                                                                                                                                                                                                                                         
 30 ) -> Result<BatteryStationChainSpec, String> {                                                                                                                                                                                                                                                                           
...                                                                                                                                                                                                                                                          
  8         #[cfg(feature = "parachain")]                                                                                                                                                                                                                                                                                    
  7         crate::chain_spec::Extensions {                                                                                                                                                                                                                                                                                  
  6             relay_chain: "rococo".into(),                                                                                                                                                                                                                                                                                
  5             parachain_id: parachain_id.into(),                                                                                                                                                                                                                                                                           
  4         },                                                                                                                                                                                                                                                                                                               
  3         #[cfg(not(feature = "parachain"))]                                                                                                                                                                                                                                                                               
  2         Default::default(),                                                                                                                                                                                                                                                                                              
  1     ))                                                                                                                                                                                                                                                                                                                   
153 }  
```
For others it is possible as they don't require parameter for parachain_id instead they use constant value 
https://github.com/AcalaNetwork/Acala/blob/master/node/service/src/chain_spec/acala.rs#L97

Reply:
Based on [this](https://github.com/paritytech/subport/issues/482#issuecomment-1311459137) let us try to find a way to be as close as possible to the parachain template code. Because currently the parachain node doesn't sync, we should try to use the new `ExportGenesisState` and `ExportGenesisWasm` method to generate our new genesis configuration for onboarding it on the rococo relaychain.